src/frameworks/next/utils.ts

```diff
@@ -209,25 +212,80 @@ export async function isUsingMiddleware(dir: string, isDevMode: boolean): Promis
 /**
  * Whether image optimization is being used
  *
- * @param dir path to `distDir` - where the manifests are located
+ * @param dir project directory
+ * @param distDir path to `distDir` - where the manifests are located
  */
-export async function isUsingImageOptimization(dir: string): Promise<boolean> {
-  let { isNextImageImported } = await readJSON<ExportMarker>(join(dir, EXPORT_MARKER));
+export async function isUsingImageOptimization(
+  projectDir: string,
+  distDir: string
+): Promise<boolean> {
+  let { isNextImageImported } = await readJSON<ExportMarker>(join(distDir, EXPORT_MARKER));
   // App directory doesn't use the export marker, look it up manually
-  if (!isNextImageImported && isUsingAppDirectory(dir)) {
-    isNextImageImported = (await readFile(join(dir, "server", "client-reference-manifest.js")))
-      .toString()
-      .includes("node_modules/next/dist/client/image.js");
+  if (!isNextImageImported && isUsingAppDirectory(distDir)) {
+    if (await isUsingNextImageInAppDirectory(projectDir, distDir)) {
+      isNextImageImported = true;
+    }
   }
 
   if (isNextImageImported) {
-    const imagesManifest = await readJSON<ImagesManifest>(join(dir, IMAGES_MANIFEST));
+    const imagesManifest = await readJSON<ImagesManifest>(join(distDir, IMAGES_MANIFEST));
     return !imagesManifest.images.unoptimized;
   }
 
   return false;
 }
 
+/**
+ * Whether next/image is being used in the app directory
+ */
+async function isUsingNextImageInAppDirectory(
+  projectDir: string,
+  nextDir: string
+): Promise<boolean> {
+  function fileIncludesNextImageComponent(fileContents: string): boolean {
+    return fileContents.includes("node_modules/next/dist/client/image");
+  }
+
+  // Note: casting the types here as this function is called after the Next.js build is done,
+  // so Next.js was already identified as a dependency. In this case getNextVersion
+  // will always return a valid string
+  const nextVersion = getNextVersion(projectDir) as string;
+  const nextVersionSemver = minVersion(nextVersion) as SemVer;
+
+  // Note: canary Next.js versions e.g. 13.4.10-canary.0 are identified as lower than 13.4.10 by SemVer.
+  // Here we use only the major, minor and patch versions for comparison.
+  const nextVersionString = `${nextVersionSemver.major}.${nextVersionSemver.minor}.${nextVersionSemver.patch}`;
```

Comment:
Create a util method if one does not exist already.
Pretty sure I saw similar logic in other places.

`getNextReleaseVersion(projectDir)`

Reply:
I found a way better way to do this using `semver`'s `coerce` , so I moved this logic to `getNextVersion` and added unit tests for that